build-tools/scripts/gettextdomains

```diff
@@ -32,13 +32,13 @@ function get_domains_and_err()
     fi
 
     for F in $SRCFILES; do
-        # strip comments from the files and match [_]_( "..." )
+        # strip comments from the files and match [_]_( "..." ), handle Ruby string interpolation
         # 1. perl: strip one-line-comments
         # 2. perl: match for _( "...") and __("..." ) (multiline too)
 	# problems left:
 	# - false matches of comments inside strings
 	# - uncaught _( at line beginning
-        MATCH=`perl -n -e 'print "$ARGV: $_" if not /(\/\/|#)/' $F | \
+        MATCH=`perl -n -e 'print "$ARGV: $_" if not /(\/\/|#[^\{])/' $F | \
```

Comment:
why not simply adding stuff like `^\s*#` ? so it match ruby definition?

Reply:
This script is also used for C++, Perl and (dropped) YCP files, it's not only for Ruby. We must be careful to not break the other languages...

Reply:
Actually, this can still fail in other ways:
It will still completely ignore lines that have a line-end comment. If you only have one or two messages, and each of them has an end-of-line comment, they are removed with this, and there will be no translation marker left, so the file is still not picked up by y2makepot.

It's better than before, but it might still fail in this subtle (or not-so-subtle) way.

Reply:
Maybe it would help to simply strip off comments first:

`s:#.*$::g`
`s://.*$::g`

Then the messages might get corrupted in this output, but that's irrelevant; its job is just to check if there are any translations, not to correctly extract them.